src/App.tsx

```diff
@@ -1,61 +1,117 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/user';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUserPosts } from './api/post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<[] | Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const noPosts = selectedUser && !posts.length && !isLoading;
```

Comment:
Better naming: `hasPosts`, `isPostsExist`
```suggestion
  const noPosts = selectedUser && !posts.length && !isLoading;
```